contracts/tokenomics/generator/src/contract.rs

```diff
@@ -365,12 +372,29 @@ fn process_after_update(deps: DepsMut, env: Env) -> Result<Response, ContractErr
                     account,
                     amount,
                 } => withdraw(deps, env, lp_token, account, amount),
+                ExecuteOnReply::SetTokensPerBlock { amount } => {
+                    set_tokens_per_block(deps, env, amount)
+                }
             }
         }
         None => Ok(Response::default()),
     }
 }
 
+fn set_tokens_per_block(
+    mut deps: DepsMut,
+    env: Env,
+    amount: Uint128,
+) -> Result<Response, ContractError> {
+    mass_update_pools(deps.branch(), env)?;
+    CONFIG.update::<_, ContractError>(deps.storage, |mut v| {
+        v.tokens_per_block = amount;
+        Ok(v)
+    })?;
+    Ok(Response::new()
+        .add_event(Event::new("Set tokens per block").add_attribute("amount", amount)))
```

Comment:
events with spaces do not work, let's just do like this
Ok(Response::new().add_attribute("action", "set_tokens_per_block"))